src/views/ProtestPage.js

```diff
@@ -0,0 +1,332 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components/macro';
+import { useHistory, useParams } from 'react-router-dom';
+import { fetchProtest, updateProtest } from '../api';
+import { Map, TileLayer, Marker } from 'react-leaflet';
+import { ProtestForm } from '../components';
+import { Switch, Route } from 'react-router-dom';
+import SocialButton, { Button } from '../components/Button/SocialButton';
+
+const mobile = `@media (max-width: 500px)`;
+
+function useFetchProtest() {
+  const [protest, setProtest] = useState(null);
+  const { id } = useParams();
+
+  useEffect(() => {
+    async function _fetchProtest(id) {
+      const result = await fetchProtest(id);
+      if (result) {
+        setProtest(result);
+      } else {
+        // TODO: handle 404
+      }
+    }
+
+    _fetchProtest(id);
+  }, [id]);
+
+  return protest;
+}
+
+function ProtestPageContent({ protest }) {
+  const history = useHistory();
+
+  const { coordinates, whatsAppLink, telegramLink, displayName, streetAddress, notes } = protest;
+
+  return (
+    <ProtestPageContainer>
+      <MapWrapper center={{ lat: coordinates.latitude, lng: coordinates.longitude }} zoom={14}>
+        <TileLayer
+          attribution='&amp;copy <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+        />
+        <Marker position={{ lat: coordinates.latitude, lng: coordinates.longitude }}></Marker>
+      </MapWrapper>
+
+      <ProtestContainer>
+        <Info>
```

Comment:
Not sure about the naming convention here. Isn't it too general? Why not `ProtestPageInfo`? (I'm kinda new with `styled-components` so not sure what's the best practices with it)

Reply:
Actually, I usually write names like ProtestPageInfo, but this time I tried going for "lighter" names. 
I think the if everything is ProtestPageSomething it becomes very busy and less readable.
I think because it's just simple markup and css it's ok to be general, but if you want to change it that's cool to.

Reply:
I'm with you for this one. Let's keep it simple